web/web.go

```diff
@@ -0,0 +1,771 @@
+	case "pg_dump", "postgres":
+		ty, issues = toSpannerTypePostgres(srcCol.Type.Name, newType, srcCol.Type.Mods)
+	default:
+		http.Error(w, fmt.Sprintf("Driver : '%s' is not supported", app.driver), http.StatusBadRequest)
+		return
+	}
+	if len(srcCol.Type.ArrayBounds) > 1 {
+		ty = ddl.Type{Name: ddl.String, Len: ddl.MaxLength}
+		issues = append(issues, internal.MultiDimensionalArray)
+	}
+	if srcCol.Ignored.Default {
+		issues = append(issues, internal.DefaultValue)
+	}
+	if srcCol.Ignored.AutoIncrement {
+		issues = append(issues, internal.AutoIncrement)
+	}
+	if len(issues) > 0 {
+		app.conv.Issues[srcTableName][srcCol.Name] = issues
+	}
+	ty.IsArray = len(srcCol.Type.ArrayBounds) == 1
+	tempColDef := sp.ColDefs[newColName]
+	tempColDef.T = ty
+	sp.ColDefs[newColName] = tempColDef
+	app.conv.SpSchema[table] = sp
+}
+
+func updateNotNull(notNullChange, table, newColName string) {
+	sp := app.conv.SpSchema[table]
+	switch notNullChange {
+	case "ADDED":
+		spColDef := sp.ColDefs[newColName]
+		spColDef.NotNull = true
+		sp.ColDefs[newColName] = spColDef
+	case "REMOVED":
+		spColDef := sp.ColDefs[newColName]
+		spColDef.NotNull = false
+		sp.ColDefs[newColName] = spColDef
+	default:
+		// We skip this.
+	}
+	app.conv.SpSchema[table] = sp
+}
+
+// Actions to be performed on a column.
+// (1) Removed: true/false
+// (2) Rename: New name or empty string
+// (3) PK: "ADDED", "REMOVED" or ""
+// (4) NotNull: "ADDED", "REMOVED" or ""
+// (5) ToType: New type or empty string
+type updateCol struct {
+	Removed bool   `json:"Removed"`
+	Rename  string `json:"Rename"`
+	PK      string `json:"PK"`
+	NotNull string `json:"NotNull"`
+	ToType  string `json:"ToType"`
+}
+type updateTable struct {
+	UpdateCols map[string]updateCol `json:"UpdateCols"`
+}
+
+func updateTableSchema(w http.ResponseWriter, r *http.Request) {
+	reqBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Body Read Error : %v", err), http.StatusInternalServerError)
+		return
+	}
+	var t updateTable
+	table := r.FormValue("table")
+	err = json.Unmarshal(reqBody, &t)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("Request Body parse error : %v", err), http.StatusBadRequest)
+		return
+	}
+	srcTableName := app.conv.ToSource[table].Name
+	for colName, v := range t.UpdateCols {
+		if v.Removed {
+			removeColumn(table, colName, srcTableName)
+			continue
+		}
+		newColName := colName
+		if v.Rename != "" && v.Rename != colName {
+			renameColumn(v.Rename, table, colName, srcTableName)
+			newColName = v.Rename
+		}
+		if v.PK != "" {
+			pkChanged(v.PK, table, newColName)
+		}
+		if v.ToType != "" {
+			updateType(v.ToType, table, newColName, srcTableName, w)
+		}
+		if v.NotNull != "" {
+			updateNotNull(v.NotNull, table, newColName)
+		}
+	}
+	app.conv.AddPrimaryKeys()
+	w.WriteHeader(http.StatusOK)
+	json.NewEncoder(w).Encode(app.conv)
+}
+
+func rateSchema(cols, warnings int64, missingPKey bool) string {
```

Comment:
This is very similar to rateSchema in report.go (except it returns colours instead of GOOD, OK, POOR etc.
Can't think of a good way to consolidate it though ...

Reply:
Let's leave as is.